packages/core/src/utils.ts

```diff
@@ -526,9 +526,9 @@ export function isObservable<T>(
 export const symbolObservable = (() =>
   (typeof Symbol === 'function' && Symbol.observable) || '@@observable')();
 
-export function isMachine(value: any): value is StateMachine<any, any, any> {
+export function isMachineNode(value: any): value is MachineNode<any, any, any> {
   try {
-    return '__xstatenode' in value;
+    return '__xstatenode' in value && value.parent === undefined;
```

Comment:
if this `isMachineNode` function is supposed to only return true for MachineNodes then we should be able to move this `__xstatenode` property onto MachineNode class and remove it from other classes (and optionally rename this property then). I dont think we really need to have this special property for state nodes, do we?